pinc/unicode.inc

```diff
@@ -40,29 +59,48 @@ function utf8_filter_to_codepoints($string, $valid_codepoints, $replacement="")
 
 # Filter out any characters in $string that are in $remove_codepoints
 function utf8_filter_out_codepoints($string, $remove_codepoints, $replacement="")
+{
+    $char_class = build_codepoint_regex_filter($remove_codepoints);
+    if(!$char_class)
+        return $string;
+
+    $pattern_string = "/[$char_class]/u";
+    return preg_replace($pattern_string, $replacement, $string);
+}
+
+# Take an array of Unicode codepoints (U+####), codepoint ranges
+# (U+####-U+####), or combined characters (U+####>U+####) and
+# return a regular expression character class to match them.
+function build_codepoint_regex_filter($codepoints)
 {
     $char_class = "";
-    foreach($remove_codepoints as $codepoint)
+    $alternatives = [];
+    foreach($codepoints as $codepoint)
     {
-        if(strpos($codepoint, '-') === False)
+        if(strpos($codepoint, '-') !== False)
         {
-            $char_class .= utf8_chr($codepoint);
+            # we have a character range
+            list($start, $end) = explode('-', $codepoint);
+            $char_class .= utf8_chr($start) . '-' . utf8_chr($end);
```

Comment:
I suppose we should really preg_quote every character we add to char_class and alternatives[] although all the regexp characters are inside a basic latin range so they will never appear explicitly.

Reply:
That's a valid concern and worth thinking about.